library/core/src/array/mod.rs

```diff
@@ -426,6 +427,18 @@ impl<T: Clone, const N: usize> Clone for [T; N] {
     }
 }
 
+#[unstable(feature = "random", issue = "130703")]
+impl<T: Random, const N: usize> Random for [T; N] {
+    fn random(source: &mut (impl RandomSource + ?Sized)) -> Self {
+        let mut buf = [const { MaybeUninit::uninit() }; N];
+        for elem in &mut buf {
+            elem.write(T::random(source));
+        }
+        // SAFETY: all elements of the array were initialized.
+        unsafe { mem::transmute_copy(&buf) }
```

Comment:
Here's a version that uses `fill_bytes()`.
```suggestion
        let mut buf = [const { MaybeUninit::<T>::uninit() }; N];
        // SAFETY: this initializes every element in the buffer with random bytes.
        unsafe {
            let slice = core::slice::from_raw_parts_mut(&raw mut buf as *mut u8, N * mem::size_of::<T>());
            source.fill_bytes(slice);
            mem::transmute_copy(&buf)
        }
```

Reply:
This is unsound for generic Ts. There's no guarantee that a Type's `random` impl shovels those bytes into their memory representation. A trivial example would be `NonZero<u8>` which could implement `Random` via rejection sampling. Setting it to 0 would be unsound.

Reply:
@the8472 I did notice that, so we would need to restrict this to types that can hold arbitrary bytes (which currently includes every type implementing `Random` except `bool`). Otherwise, use a slower custom implementation with rejection sampling or modulo.